lib/ansible/modules/cloud/amazon/s3_cors.py

```diff
@@ -0,0 +1,225 @@
+  returned: always
+  type: string
+  sample: 'bucket-name'
+rules:
+  description: list of current rules
+  returned: always
+  type: list
+  sample: [
+     {
+        "allowed_headers": [
+          "Authorization"
+        ],
+        "allowed_methods": [
+          "GET"
+        ],
+        "allowed_origins": [
+          "*"
+        ],
+        "max_age_seconds": 30000
+      }
+    ]
+'''
+
+import re
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (boto3_conn, ec2_argument_spec, get_aws_connection_info,
+                                      snake_dict_to_camel_dict, camel_dict_to_snake_dict)
+
+
+def create_or_update_bucket_cors(connection, module):
+
+    name = module.params.get("name")
+    rules = module.params.get("rules", [])
+    changed = False
+
+    try:
+        current_camel_rules = connection.get_bucket_cors(Bucket=name)['CORSRules']
+    except ClientError as e:
+        current_camel_rules = []
+
+    new_camel_rules = snake_dict_to_camel_dict(rules)
+
+    if not (len(new_camel_rules) == len(current_camel_rules)):
+        changed = True
+
+    if not changed:
+        for rule_index, new_camel_rule in enumerate(new_camel_rules):
+            if not changed:
+                if not (new_camel_rule.keys() == current_camel_rules[rule_index].keys()):
+                    changed = True
+                    break
+                for rule_key in new_camel_rule.keys():
+                    if rule_key == 'MaxAgeSeconds':
+                        if not (new_camel_rule[rule_key] == current_camel_rules[rule_index].get(rule_key, None)):
+                            changed = True
+                            break
+                    else:
+                        if not (set(new_camel_rule[rule_key]) == set(current_camel_rules[rule_index].get(rule_key, []))):
+                            changed = True
+                            break
+            else:
+                break
+
+    if changed:
+        try:
+            cors = connection.put_bucket_cors(Bucket=name, CORSConfiguration={'CORSRules': new_camel_rules})
+        except ClientError as e:
+            module.fail_json(
+                msg="Error putting bucket CORS",
+                exception=traceback.format_exc(),
+                **camel_dict_to_snake_dict(e.message)
+            )
+
+    module.exit_json(changed=changed, name=name, rules=rules)
+
+
+def destroy_bucket_cors(connection, module):
+
+    name = module.params.get("name")
+    changed = False
+
+    try:
+        cors = connection.delete_bucket_cors(Bucket=name)
+        changed = True
+    except ClientError as e:
+        module.fail_json(
+            msg="Error deleting bucket CORS",
+            exception=traceback.format_exc(),
+            **camel_dict_to_snake_dict(e.message)
```

Comment:
This looks like you've copied the boto example rather than the boto3 example.

Change `e.message` to `e.response` and maybe change `msg="Error deleting bucket CORS"` to `msg=e.message`.